### ISSUE TITLE
ModuleNotFoundError from ad-ench applies here, too

### ISSUE THREAD

**Issue by chbndrhnns** [benleb/ad-automoli#15]
I am seeing the [`ModuleNotFoundError`](https://github.com/benleb/ad-ench/issues/1) here, too:

```
2020-01-20 15:32:48.768224 WARNING livingroom: Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/appdaemon/app_management.py", line 820, in check_app_updates
    await self.init_object(app)
  File "/usr/lib/python3.8/site-packages/appdaemon/app_management.py", line 271, in init_object
    modname = await utils.run_in_executor(self, __import__, app_args["module"])
  File "/usr/lib/python3.8/site-packages/appdaemon/utils.py", line 276, in run_in_executor
    response = future.result()
  File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/appdaemon/apps/automoli/automoli.py", line 32, in <module>
    from adutils import ADutils, hl
ModuleNotFoundError: No module named 'adutils'
```

**Comment by benleb** [comment]
Hm how do you use automoli? via hacs? python venv for appdaemon? docker?

appdaemon scans the apps directories for requirements.txt files and should install adutils itself but that seems not to work in this case!?

**Comment by chbndrhnns** [comment]
I am using it via hacs. Appdaemon is running via hassio. After adding ad-util to the python packages config section of the appdaemon config in Hassio, it works.

**Comment by benleb** [comment]
Hm interesting, afaik you are not the only user with such a setup and it works for others... But if it works for you now, nice! I will leave this open for a few days, if someone other expects the same, please comment that we can track down and solve the problem! :)

**Comment by benleb** [comment]
> I am using it via hacs. Appdaemon is running via hassio. After adding ad-util to the python packages config section of the appdaemon config in Hassio, it works.

@gogui63 this workaround from @chbndrhnns should work for you too. I will check whats needed to solve this in a proper way...

**Comment by benleb** [comment]
Ok after talking to @frenck, which maintains the addon, it seems that I can not do much for you in this case. From my perspective (at least currently) this seems to be an issue in the Hassio addon as it works in standalone/Docker AppDaemon setups without problems.

Suggesting you to open an issue in the Addon repo https://github.com/hassio-addons/addon-appdaemon/issues

**Comment by benleb** [comment]
After digging a little deeper (with assistance from @frenck), it seems the correct and intended way to go for hassio users is to use https://github.com/hassio-addons/addon-appdaemon#option-python_packages

I will document this in the readme... I am aware that it's not the most convenient way to manually add a python package, but for now, it is the only supported one :)

**Comment by gogui63** [comment]
Yes , that's work for me ! Thanks

**Comment by benleb** [comment]
Should be fixed in [v0.5.4](https://github.com/benleb/ad-automoli/releases/tag/v0.5.4) (hopefully)

Can anyone confirm this?

**Comment by chbndrhnns** [comment]
With 0.5.4, I see this error message after updating. If I restart AppDaemon, though, there is no error message. Maybe the installation happens only on boot of AppDaemon?

```
2020-02-13 16:03:49.133312 INFO AppDaemon: App initialization complete
2020-02-13 16:04:42.829624 INFO AppDaemon: Removing module /config/appdaemon/apps/automoli/__init__.py
2020-02-13 16:04:42.831479 INFO AppDaemon: Removing module /config/appdaemon/apps/automoli/automoli.py
2020-02-13 16:04:42.832017 INFO AppDaemon: Removing module /config/appdaemon/apps/automoli/pkg_helper.py
2020-02-13 16:04:42.832954 INFO AppDaemon: Terminating stairs
2020-02-13 16:04:44.900973 WARNING AppDaemon: No app description found for: /config/appdaemon/apps/automoli/__init__.py - ignoring
2020-02-13 16:04:44.903679 INFO AppDaemon: Loading App Module: /config/appdaemon/apps/automoli/automoli.py
installing required packages: {'adutils~=0.4.9'}
{'adutils~=0.4.9'} successfully installed!
2020-02-13 16:04:46.606188 WARNING Error: ------------------------------------------------------------
2020-02-13 16:04:46.607930 WARNING Error: Unexpected error loading module: /config/appdaemon/apps/automoli/automoli.py:
2020-02-13 16:04:46.608726 WARNING Error: ------------------------------------------------------------
2020-02-13 16:04:46.611639 WARNING Error: Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/appdaemon/app_management.py", line 788, in check_app_updates
    await utils.run_in_executor(self, self.read_app, mod["name"], mod["reload"])
  File "/usr/lib/python3.8/site-packages/appdaemon/utils.py", line 276, in run_in_executor
    response = future.result()
  File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/lib/python3.8/site-packages/appdaemon/app_management.py", line 580, in read_app
    self.modules[module_name] = importlib.import_module(module_name)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/config/appdaemon/apps/automoli/automoli.py", line 69, in <module>
    from adutils import ADutils, hl
ModuleNotFoundError: No module named 'adutils'

2020-02-13 16:04:46.613837 WARNING Error: ------------------------------------------------------------
2020-02-13 16:04:46.614657 WARNING AppDaemon: Removing associated apps:
2020-02-13 16:04:46.615425 WARNING AppDaemon: stairs
2020-02-13 16:04:46.619187 WARNING AppDaemon: No app description found for: /config/appdaemon/apps/automoli/pkg_helper.py - ignoring
```

**Comment by benleb** [comment]
thanks @chbndrhnns! I think i got the problem and released a fixed [v0.5.5](https://github.com/benleb/ad-automoli/releases/tag/v0.5.5)

Can anyone confirm this now? :D

**Comment by chbndrhnns** [comment]
Seems to work now. Thanks so much for troubleshooting this!
